src/lwc/geFormField/geFormField.js

```diff
@@ -46,12 +46,30 @@ export default class GeFormField extends LightningElement {
     }
 
     connectedCallback() {
-        const { defaultValue } = this.element;
-        if(defaultValue) {
+        const { defaultValue, recordValue } = this.element;
+
+        if(recordValue) {
+
+            // set the record value to the element value
+            this.value = recordValue;
+        } else if(defaultValue) {
+           
+            // Set the default value if there is one
+            // and no record value. 
             this.value = defaultValue;
         }
     }
 
+    handleValueChange(event) {
+        this.value = this.getValueFromChangeEvent(event);
+        window.clearTimeout(this.changeTimeout);
+        this.changeTimeout = setTimeout(() => {
```

Comment:
Is this setTimeout() relying on a fixed delay? Is there a more event-driven way to handle this? I believe this usage is generally frowned on

Reply:
Good catch, this whole function can get removed. Line 37 in this file already defines `handleValueChange` using the utilCommon debounce function.